pkg/3scale/amp/operator/base_apimanager_logic_reconciler.go

```diff
@@ -106,11 +113,17 @@ func (r *BaseAPIManagerLogicReconciler) ReconcilePrometheusRules(desired *monito
 	if err != nil {
 		return err
 	}
-	if !kindExists {
-		r.Logger().Info("Install prometheus-operator in your cluster to create prometheusrules objects", "Error creating prometheusrule object", desired.Name)
+
+	if !r.apiManager.IsMonitoringEnabled() && !kindExists {
 		return nil
 	}
 
+	if r.apiManager.IsMonitoringEnabled() && !kindExists {
+		err = fmt.Errorf("Error creating prometheusrule object '%s'", desired.Name)
+		r.Logger().Error(err, "Install prometheus-operator in your cluster to create prometheusrules objects")
+		return err
+	}
+
```

Comment:
Before we always logged the message when the kind didn't exist, independently on whether monitoring was enabled or not, and we logged the message as Info type.
Now we only log that message when monitoring is enabled and the kind doesn't exist, and in that case we log as error and return an error instead of nil. However, I'm not entirely sure about this last behavior. If monitoring is enabled and the kind doesn't exist, should we stop processing the next resources?